### DISCUSSION TITLE
Copilot more effective given your code

### DISCUSSION THREAD

**Discussion by VideoCarp** [community/community#15906] [Copilot Conversations]
Copilot depends on the code that's available publicly and that it is provided. Naturally, it should be pretty bad for languages like F#. F# is decently known, but perhaps not enough for copilot to generate meaningful code. Python's quite famous, and when I ask GitHub copilot to write a lexer, it has a blank. And when I tell it to do anything in an empty F# file, it decides I'm either writing Go or C#. But after I wrote 10 or 20 lines and defined a few meaningful variable names (at least to me), copilot could predict exactly what I'm going to write next for some cases. It looks like GitHub copilot works much better when you give  it your own code. All the files publicly available do have a significant difference. It knew the next step in my program before I could think of it.

But it only learnt this after my code showed it the syntax of the language, basic functions and my norms.

![download](https://user-images.githubusercontent.com/66365570/166215595-264d0c08-c276-4ee5-a7df-3a12f7f4f38f.png)

**Comment by Urethramancer** [comment]
I got the spooky prediction too, after seeding projects with some base code. In one case, I started blank (main() and database opening), wrote some SQL statements to set up tables (which Copilot also helped with pretty well), then switched to the DB func. It immediately suggested the right thing, just with the wrong DB access package. Easy fix, and after that it literally wrote all my get/set functions for the tables.

It's sometimes amazing at picking up syntax of obscure languages though. I also wrote a package this weekend which it not only picked up usage of immediately, but also started suggesting in other projects, where relevant. It's borderline creepy!
